### PR TITLE
Added APIProduct Store REST API basic flows

### PR DESCRIPTION
**Pull request by abimarank**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+ *
+ */
+
+package org.wso2.carbon.apimgt.rest.api.store.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.wso2.carbon.apimgt.rest.api.store.dto.APIBusinessInformationDTO;
+
+import io.swagger.annotations.*;
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java]: imports with stars are not the practice

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+import com.fasterxml.jackson.annotation.*;
+
+import javax.validation.constraints.NotNull;
+
+
+
+
+
+@ApiModel(description = "")
+public class APIProductDTO  {
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java]: Class comment would be nice.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+  public String getVersion() {
+    return version;
+  }
+  public void setVersion(String version) {
+    this.version = version;
+  }
+
+  
+  /**
+   **/
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java]: methods comments should be added

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/ProductsApiService.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+
+/*
+ *
+ *  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
```
Comment (sajinidesilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/ProductsApiService.java]: Wrong format of copyright header. Extra star is added to every line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java (hunk 1 above)
Comment (abimarank) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/dto/APIProductDTO.java]: Hi, Thanks for the suggestion, this code was generated based on YAML configuration, anyway I will fix it.

**Pull request merged**